File: tests/src/test/java/com/github/klee0kai/stone/test/bindinstance/BindInstanceTests.java
package com.github.klee0kai.stone.test.bindinstance;

import com.github.klee0kai.stone.Stone;
import com.github.klee0kai.test.Application;
import com.github.klee0kai.test.Context;
import com.github.klee0kai.test.bindinstance.di.BindInstanceTextComponent;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class BindInstanceTests {


    @Test
    public void simpleInitTest() {
        Application application = Application.create();
        Context context = Context.create();
        BindInstanceTextComponent DI = Stone.createComponent(BindInstanceTextComponent.class);
        DI.bind(application, context);

        assertNotEquals(DI.appModule().application().uuid, DI.appModule().context().uuid);
    }

    @Test
    public void simpleInitTest2() {
        Application application = Application.create();
        Context context = Context.create();
        BindInstanceTextComponent DI = Stone.createComponent(BindInstanceTextComponent.class);
        DI.bind(context, application);
        assertNotEquals(DI.appModule().application().uuid, DI.appModule().context().uuid);
    }


    @Test
    public void simpleInitTest3() {
        Application application = Application.create();
        Context context = Context.create();
        BindInstanceTextComponent DI = Stone.createComponent(BindInstanceTextComponent.class);
        DI.bind(application);

        assertNull(DI.appModule().context());
        assertNotNull(DI.appModule().application());
    }

    @Test
    public void simpleInitTest4() {
        Application application = Application.create();
        Context context = Context.create();
        BindInstanceTextComponent DI = Stone.createComponent(BindInstanceTextComponent.class);
        DI.bind(context);

        assertNotNull(DI.appModule().context());
        assertNull(DI.appModule().application());
    }

}

File: test_kotlin/src/test/kotlin/com/github/klee0kai/test/bindinstance/BindInstanceTests.kt
package com.github.klee0kai.test.bindinstance

import com.github.klee0kai.stone.Stone
import com.github.klee0kai.test.Application
import com.github.klee0kai.test.Context
import com.github.klee0kai.test.bindinstance.di.BindInstanceTextComponent
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class BindInstanceTests {
    @Test
    fun simpleInitTest() {
        val application = Application.create()
        val context = Context.create()
        val DI = Stone.createComponent(
            BindInstanceTextComponent::class.java
        )
        DI.bind(application, context)
        Assertions.assertNotEquals(
            DI.appModule().application()!!.uuid, DI.appModule().context()!!.uuid
        )
    }

    @Test
    fun simpleInitTest2() {
        val application = Application.create()
        val context = Context.create()
        val DI = Stone.createComponent(
            BindInstanceTextComponent::class.java
        )
        DI.bind(context, application)
        Assertions.assertNotEquals(
            DI.appModule().application()!!.uuid, DI.appModule().context()!!.uuid
        )
    }

    @Test
    fun simpleInitTest3() {
        val application = Application.create()
        val context = Context.create()
        val DI = Stone.createComponent(
            BindInstanceTextComponent::class.java
        )
        DI.bind(application)
        Assertions.assertNull(DI.appModule().context())
        Assertions.assertNotNull(DI.appModule().application())
    }

    @Test
    fun simpleInitTest4() {
        val application = Application.create()
        val context = Context.create()
        val DI = Stone.createComponent(
            BindInstanceTextComponent::class.java
        )
        DI.bind(context)
        Assertions.assertNotNull(DI.appModule().context())
        Assertions.assertNull(DI.appModule().application())
    }
}